core/preconditioner/isai.cpp

```diff
@@ -97,39 +98,79 @@ convert_to_csr_and_sort(std::shared_ptr<const Executor> &exec, const LinOp *mtx,
     if (!skip_sorting) {
         copy->sort_by_column_index();
     }
-    return {copy.release(), std::default_delete<const Csr>{}};
+    return copy;
```

Comment:
The intel compiler does not like the implicit conversion from `unique_ptr` to `shared_ptr` here (see [the pipeline for intel](https://gitlab.com/ginkgo-project/ginkgo-public-ci/pipelines/139151360)).
You could use the [(13.) constructor of shared_ptr](https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr), which uses an rvalue reference to a `unique_ptr` as a parameter. That should resolve the compilation issue.
```suggestion
    return {std::move(copy)};
```

Reply:
Alternatively, you can also generate `copy` as a `shared_ptr` (do the implicit conversion there):
```c++
auto copy = gko::share(Csr::create(exec));
```
or
```c++
std::shared_ptr<Csr> copy = Csr::create(exec);
```